### PR TITLE
Run test

### PR DESCRIPTION
**Pull request by matklad**
Hi!

This is a proof of concept for https://github.com/rust-lang-nursery/rls-vscode/issues/225, which, at least in my experience, is a **huge** productivity boost.

The current implementation approach here risks summoning Zalgo, I'll look into using save analysis data to detect available test and Cargo metadata to find out the proper cargo comman-line for this. OTOH, even this simplistic approach covers 80% of use-cases, so I think merging this almost as is (with some regex tweaking) might be a good idea as well.

### CODE REVIEW

**Comment by matklad** [comment]
Here's how the UX looks: https://youtu.be/Lntgjpk18sk

**Comment by matklad** [comment]
Apparently, the proper VS Code UI for this is code lens. Looking into implementing it!

File: package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
         "theme": "dark"
     },
     "engines": {
-        "vscode": "^1.22.0"
+        "vscode": "^1.24.0"
```
[the remark this answers is not in the document]
Comment (nrc) [reply package.json]: Latest stable and insiders only for now

**Comment by matklad** [comment]
Seems to work:

image

:)

**Comment by matklad** [comment]
Updated.

**Comment by matklad** [comment]
The RLS-side of this has been merged

**Comment by nrc** [comment]
Thanks!

**Comment by sfackler** [comment]
The "Run test" button things have appeared in vscode with the latest nightly, but they don't seem to do anything at all - no debug logs from rls at all. Is there any explicit configuration that's needed?

**Comment by matklad** [comment]
No, no config should be necessary. If you see the lens, that means that RLS side of things works, and the problem is within VS Code extension.

**Comment by matklad** [comment]
Looks like https://github.com/Microsoft/vscode/issues/53673

**Comment by matklad** [comment]
Fixed by https://github.com/rust-lang-nursery/rls-vscode/pull/361

**Comment by MajorBreakfast** [comment]
Whoa! This is great! I just saw the "Run test" button appear under one of my `#[test]` annotations. Thx! Brilliant idea 🎉👍👍

**Comment by matklad** [comment]
@MajorBreakfast thanks! Thers’s still a ton of work to do to make this work properly :-)

**Pull request merged**